### ISSUE TITLE
PT019 `pytest-fixture-param-without-value` misses public fixture

### ISSUE THREAD

**Issue by mschoettle** [astral-sh/ruff#17132]
### Summary

I defined a fixture in `conftest.py` and I was expecting a violation since it satisfies the same logic as described in the [rule description.](https://docs.astral.sh/ruff/rules/pytest-fixture-param-without-value/).

```python
# in conftest.py
def my_fixture() -> None:
    pass
```

```python
def some_test(my_fixture: None):
    pass
```

### Version

ruff 0.11.2 (4773878ee 2025-03-21)

**Comment by dhruvmanila** [comment]
Looking at the code, it seems that you'd need to prefix the fixture with an underscore, so `_my_fixture` works, which I think would be to match the upstream rule (?).

So, the following would work:
```py
def test_some(_my_fixture: None):
    pass
```

You'd also need to use `test_` prefix and not suffix.

**Comment by mschoettle** [comment]
It does work when prefixing with an underscore. When reading the rationale ([upstream](https://github.com/m-burst/flake8-pytest-style/blob/master/docs/rules/PT019.md) and `ruff`, i.e., unused fixture in the test function) it seems to me that it should apply to any fixture that is not used in the test function, so also those not prefixed. Unless fixtures should be prefixed all the time? 🤔 

> You'd also need to use test_ prefix and not suffix.

Good catch. It was just in this minimal example.

**Comment by MichaReiser** [comment]
The rule uses heuristics to detect fixtures, and we need to balance false positives (where Ruff detects something as a fixture where it isn't) with false negatives. That's why it requires the underscore.

**Comment by manueljacob** [comment]
In the example, if the `test_something` function was not empty, rule ARG001 would complain about the unused parameter.

PT019 in current form triggers if the fixture name starts with an underscore, regardless of whether the fixture value is used inside the test function or not. See also https://github.com/astral-sh/ruff/issues/18119#issuecomment-3743860839 for analysis.
